research/berry/berrylib/fast_math.py

```diff
@@ -0,0 +1,107 @@
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+
+@jax.jit
+def logdet(m):
+    """Compute the log of the determinant in a numerically stable way."""
+    if isinstance(m, np.ndarray):
+        numpy = np
+    else:
+        numpy = jnp
+    sign, logdet = numpy.linalg.slogdet(m)
+    return sign * logdet
```

Comment:
Same comment as below. This should just return logdet.

Reply:
This actually isn't even used anywhere